services/src/main/java/org/keycloak/services/resources/admin/GroupResource.java

```diff
@@ -162,9 +162,11 @@ public Stream<GroupRepresentation> getSubGroups(@QueryParam("first") @DefaultVal
         @QueryParam("briefRepresentation") @DefaultValue("false") Boolean briefRepresentation) {
         this.auth.groups().requireView(group);
         boolean canViewGlobal = auth.groups().canView();
-        return group.getSubGroupsStream(first, max)
+        return group.getSubGroupsStream(null, null)
             .filter(g -> canViewGlobal || auth.groups().canView(g))
-            .map(g -> GroupUtils.populateSubGroupCount(g, GroupUtils.toRepresentation(auth.groups(), g, !briefRepresentation)));
+            .map(g -> GroupUtils.populateSubGroupCount(g, GroupUtils.toRepresentation(auth.groups(), g, !briefRepresentation)))
+            .skip(first)
+            .limit(max);
```

Comment:
If I may, I suggest using the no parameter variant along with `StreamsUtil` that handles the pagination:

```
        return StreamsUtil.paginatedStream(
                        group.getSubGroupsStream()
                        .filter(g -> canViewGlobal || auth.groups().canView(g)), first, max)
                .map(g -> GroupUtils.populateSubGroupCount(g, GroupUtils.toRepresentation(auth.groups(), g, !briefRepresentation)));
```

Reply:
Hi @sguilhen, thanks for the comment and suggestion! Sounds good, I've implemented it.